parsers/TableOfContentsParser.py

```diff
@@ -0,0 +1,419 @@
+    possible_starts = ["CONTENTS:", "TABLE OF CONTENTS"]
+
+    @staticmethod
+    def filter_lines_by_section_keyword(lines: List[str]) -> int:
+
+        section_start_line_id = FilterLinesBySectionKeyword.get_toc_section_start(lines)
+        if section_start_line_id == -1:
+            return []
+        section_end_line_id = FilterLinesBySectionKeyword.get_first_non_empty_non_numeric_ending_line_id(lines, section_start_line_id + 1)
+        if section_end_line_id == -1:
+            return []
+
+        answer = list(filter(lambda x: len(x.strip()), lines[section_start_line_id+1:section_end_line_id]))
+        answer = list(filter(lambda x: x.strip()[-1].isnumeric(), answer))
+        answer = list(map(lambda x: x.strip(), answer))
+
+        # logger.debug(random_filename)
+        return answer
+
+    @staticmethod
+    def get_toc_section_start(lines: List[str]) -> int:
+        section_start = -1
+        # max_possible_start_len = max(map(lambda x: len(x), possible_starts))
+        line_id = 0
+
+        for line in lines:
+            line_stripped = line.strip().lower()
+            # if len(line_stripped) > max_possible_start_len + 5:
+            #     continue
+            for single_start in FilterLinesBySectionKeyword.possible_starts:
+                # if line_stripped.startswith(single_start.lower()):
+                if line_stripped.startswith(single_start.lower()):
+                    # logger.warning(line_stripped)
+                    section_start = line_id
+                    break
+            if section_start >= 0:
+                break
+            line_id += 1
+        return section_start
+
+    @staticmethod
+    def get_first_non_empty_non_numeric_ending_line_id(lines: List[str], start_line_id: int) -> int:
+        for line_id in range(start_line_id, len(lines)):
+            line_stripped = lines[line_id].strip().lower()
+            if sum(map(lambda x: x.lower() in line_stripped, FilterLinesBySectionKeyword.possible_starts)):
+                continue
+            if line_stripped == "contents":
+                continue
+            if len(line_stripped) == 0:
+                continue
+            if not line_stripped[-1].isnumeric():
+                return line_id
+        return -1
+
+
+class FilterLinesByNumWildcardNum:
+    @staticmethod
+    def filter_lines_by_num_wildcard_num(lines: List[str]) -> List[str]:
+        sep = DOTS_SEP
+        new_lines = []
+        for line in lines:
+            new_line = line.strip()
+            if len(new_line) < 2:
+                continue
+            if not (new_line[0].isnumeric() and new_line[-1].isnumeric()):
+                continue
+            new_line = new_line.replace(". . ", sep).replace(". .", sep)
+            # print(new_line)
+            new_lines.append(new_line)
+
+        return new_lines
+
+
+if False:
+    # The code here is kept so that we have it for future prototyping, but disabled so that it's not used.
```

Comment:
Leaving "disabled" code like this is bad practice, and it generates a lot of warnings. 
You can create separate git branch for this purpose.